EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -227,7 +275,14 @@ protected List<int> GetBinsToSearch(Ms2ScanWithSpecificMass scan)
         protected static int BinarySearchBinForPrecursorIndex(List<int> peptideIdsInThisBin, double peptideMassToLookFor, List<PeptideWithSetModifications> peptideIndex)
         {
             int m = 0;
+
             int l = 0;
+            int peptidesInBin = 0;
+            if (peptideIdsInThisBin.Any())
+            {
```

Comment:
I don't think you need any of this? peptideIdsInThisBin should never be empty; if it's not null it should have at least one peptide in it

Reply:
done